presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java

```diff
@@ -420,10 +420,10 @@ public PlanNode visitGroupId(GroupIdNode node, RewriteContext<Set<Symbol>> conte
         {
             ImmutableSet.Builder<Symbol> expectedInputs = ImmutableSet.builder();
 
-            Map<Symbol, Symbol> newArgumentMappings = node.getArgumentMappings().entrySet().stream()
-                    .filter(entry -> context.get().contains(entry.getKey()))
-                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
-            expectedInputs.addAll(newArgumentMappings.values());
+            List<Symbol> newAggregationArguments = node.getAggregationArguments().stream()
+                    .filter(context.get()::contains)
+                    .collect(Collectors.toList());
```

Comment:
`.forech(expectedInputs::add);`?

Reply:
I would've done it that way, but newAggregationArguments is needed below.